### PR TITLE
Let initial=-jnp.inf by default for nn.softmax and nn.log_softmax.

### PR DESCRIPTION
**Pull request by carlosgmartin**
#20626

### CODE REVIEW

File: jax/_src/nn/functions.py [hunk 1]
```diff
@@ -454,7 +454,7 @@ def glu(x: ArrayLike, axis: int = -1) -> Array:
 def log_softmax(x: ArrayLike,
                 axis: int | tuple[int, ...] | None = -1,
                 where: ArrayLike | None = None,
-                initial: ArrayLike | None = None) -> Array:
+                initial: ArrayLike | None = -jnp.inf) -> Array:
```
Comment (jakevdp) [inline jax/_src/nn/functions.py:457]: Is there any reason to still accept `None` here? Should it just be `initial: ArrayLike = -jnp.inf`?

Comment (carlosgmartin) [reply jax/_src/nn/functions.py:457]: I was thinking there might be codebases that currently pass in `None` or a variable whose type includes `None` as a possibility, which could cause their type-checking to fail.

I'd support deprecating such use, though. In fact, my inclination would be to remove the `initial` argument entirely, since I don't think there's a case where anything other than `-jnp.inf` is necessary. (Remembering that this is just to handle empty cases for the `jnp.max` inside `softmax`.)

Comment (jakevdp) [reply jax/_src/nn/functions.py:457]: Yeah, that makes sense. Let's land this as-is, then I can work on the deprecation.

**Review by jakevdp — APPROVED** [review]
